File: Variables/Java/simpleVariables.java
class Main{
    public static void main(String[] args){
        // Initializing a Variable 

        // In Java 8, Declaring Variables will require some knowledge of the Data Type that you'll be using as a value to store. This is because Java 8 unlike Python and Javascript is a Compiled Language as opposed to an Interpreted Language. 

        // We will be working with String, Integer, and Boolean Variables.

        // To Declare Variables, we will need the following:
        /* 
            1. The Keyword String, int, or boolean
                a. If the value is a String, you'll use the keyword String
                b. If the value is a Whole Number, you'll use the keyword int
                c. If the value is a Boolean, you'll use the keyword boolean
            2. A Variable Name
            3. The Assignment Operator
            4. The Value
            5. A Semi Colon
        */

        /* Initializing a String Variable in Java 8 */
        String progLanguage = "Java 8";

        /* Initializing a Whole Number in Java 8 */
        int num = 21;

        /* Initializing a Boolean Number in Java 8 */
        boolean isCoding = true;

        // Printing out the Values of your Variables to the Console using System.out.print() or System.out.println()
        System.out.println(progLanguage);
        System.out.println(num);
        System.out.println(isCoding);

        // You can only get the Data Type value of Object Type Data and not Primitive Type Data. Therefore, if we have a primitive data type that we are working with such as 'int' and 'bool', then we will have to cast it as an Object.

        // Already an Object Type
        progLanguage.getClass().getSimpleName();
        
        // Casting as Object Types
        ((Object)num).getClass().getSimpleName();
        ((Object)isCoding).getClass().getSimpleName();

        // We can store its reference in a variable: 

        // Since they are all objects we will declare them as Objects
        Object progLanguageDT = progLanguage.getClass().getSimpleName();
        Object numDT = ((Object)num).getClass().getSimpleName();
        Object isCodingDT = ((Object)num).getClass().getSimpleName();
        
        // Then we can print its value from that reference
        System.out.println(progLanguageDT);
        System.out.println(numDT);
        System.out.println(isCodingDT);

        //You can ofcourse, also choose to print directly from the reference without storing into a variable as follows: 
        System.out.println(progLanguage.getClass().getSimpleName());
        System.out.println(((Object)num).getClass().getSimpleName());
        System.out.println(((Object)isCoding).getClass().getSimpleName());

    }
}

File: Variables/JavaScript/simpleVariables.js
t

//1. A String
let progLanguage = "JavaScript";

//2. A Number 
let number = 25;

//3. A Boolean 
let isCoding = true;

/* It can also be the other two Primitive Data Types */

//4. An Undefined
let noValue; //A Variable with No Value is 'undefined'
let undef = undefined //Can explicitly make 'undefined'

//5. A Null Value
let emptyObj = null; //Null is an Object with no Reference! (Like the number 0 is a number with no Quantity.)


//Printing out the Values of your Variables to the Console using console.log()
console.log(progLanguage);
console.log(number);
console.log(isCoding);


//Create a reference to the Data Type of your Variables using the JavaScript Method typeof()
typeof(progLanguage);
typeof(number);
typeof(isCoding);

//Log to the Console The Reference to Your Variables Data Type

/* Version 1: Directly Inserting the Type Of Method Reference into the console.log() as an argument */

console.log(typeof(progLanguage));
console.log(typeof(number));
console.log(typeof(isCoding));

/* Version 2: Storing the typeof() Reference as a variable and then using the Variable as an argument for the console.log() */

//Step 1: Create the Variables and Store the Reference as a Value
let progLanguageDT = typeof(progLanguage);
let numberDT = typeof(number);
let isCodingDT = typeof(isCoding);

//Step 2: Log to the Console the Variables with the Stored Reference
console.log(progLanguageDT);
console.log(numberDT);
console.log(isCodingDT);